### PR TITLE
Update behaviour for request payload retrieval

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#514](https://github.com/ballerina-platform/ballerina-standard-library/issues/514)

## Checklist
- [X] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

**Comment by chamil321** [comment]
@ayeshLK Let's add a test case for the fixed issue

File: native/src/main/java/io/ballerina/stdlib/http/api/HttpCallableUnitCallback.java [hunk 1]
```diff
@@ -80,8 +80,8 @@ public void notifyFailure(BError result) {
     public void notifyFailure(BError error) { // handles panic and check_panic
         // This check is added to release the failure path since there is an authn/authz failure and responded
         // with 401/403 internally.
+        cleanupRequestAndContext();
```
Comment (chamil321) [inline native/src/main/java/io/ballerina/stdlib/http/api/HttpCallableUnitCallback.java:83]: Move this to L81 as above comment is applied to L84

File: ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+@test:Config {
+    groups: ["payloadRetrieveAfterRespond"]
+}
+function testPayloadRetrievalAfterRespondTest() returns error? {
+    http:Response|error response = payloadRetrievalTestClient->get("/passthrough");
+    if (response is http:Response) {
+        test:assertEquals(response.statusCode, 200, msg = "Found unexpected output");
+        assertHeaderValue(check response.getHeader(CONTENT_TYPE), TEXT_PLAIN);
+        assertTextPayload(response.getTextPayload(), "Request Processed successfully");
+        runtime:sleep(5);
```
Comment (chamil321) [inline ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal:101]:
```suggestion
```
Do we need it?

File: ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        response.setHeader("X-Type", "test");
+        check caller->respond(response);
+    }
+
+    resource function get get4XX(http:Caller caller, http:Request req) returns error? {
+        http:Response response = new;
+        response.statusCode = 400;
+        response.setTextPayload("data-binding-failed-due-to-bad-request");
+        check caller->respond(response);
+    }
```
Comment (chamil321) [inline ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal:89]: Do we use these resources?

File: ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        return "Request Processed successfully";
+    }
+}
+
+service /backend on payloadRetrievalListener {
+    resource function 'default getJson(http:Caller caller, http:Request req) returns error? {
+        http:Response response = new;
+        response.setJsonPayload({id: "chamil", values: {a: 2, b: 45, c: {x: "mnb", y: "uio"}}});
+        check caller->respond(response);
+        jsonPaylod = check req.getJsonPayload();
```
Comment (chamil321) [inline ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal:51]:
```suggestion
        requesJsonPaylod = check req.getJsonPayload();
```
Shall we prefix with `request` so the test is more readable

File: ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal (hunk 3 above)
Comment (ayeshLK) [reply ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal]: Nope

File: ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal (hunk 4 above)
Comment (ayeshLK) [reply ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal]: Updated all the places.

**Review by chamil321 — APPROVED** [review]

**Pull request merged**